pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -188,6 +259,57 @@ public long[] transformToLongValuesSV(ValueBlock valueBlock) {
     return _longValuesSV;
   }
 
+  @Override
+  public Pair<long[], RoaringBitmap> transformToLongValuesSVWithNull(ValueBlock valueBlock) {
+    int length = valueBlock.getNumDocs();
+    if (_longValuesSV == null) {
+      _longValuesSV = new long[length];
+    }
+    RoaringBitmap bitmap;
+    DataType resultDataType = getResultMetadata().getDataType();
+    switch (resultDataType.getStoredType()) {
+      case UNKNOWN:
+        bitmap = new RoaringBitmap();
+        bitmap.add(0L, length);
+        for (int i = 0; i < length; i++) {
+          _longValuesSV[i] = (long) DataSchema.ColumnDataType.LONG.getNullPlaceholder();
```

Comment:
what's the need to do copy when all bits are set to 0 in bitmap?
Is it to show some placeholder value in the end?

Reply:
I was thinking we should keep the behavior consistent when null handling option is not enabled.

Reply:
I don't think there is a plan to make nullHandling as default due to perf penalty we add for handling null. I added a comment explaining why we need the default values.